### PR TITLE
Linux Kernel 5.15

### PR DESCRIPTION
**Pull request by cthbleachbit**
Topic Description
-----------------

This PR updates main linux kenel to 5.15.4 and various other supporting packages. Here is a summary of changes for the primary architectures:

#### amd64:

- DAMON: Data Access Monitoring framework
- I2C/GPIO over VirtIO
- AMD PTDMA engine
- NTFS3 support from paragon
- AESNI/SSE accelerated SM4 cryptography routines

#### arm64:

- DAMON: Data Access Monitoring framework
- I2C/GPIO over VirtIO
- NTFS3 support from paragon
- HyperV guest modules: PCIe, balloning, input, storage, blah blah
- Apple M1 DART IOMMU support

For detailed changes, see [this spreadsheet](https://docs.google.com/spreadsheets/d/1-ohaMnTXqQXBrtbV7v4G0YWu_mzw80zTGRgeDJfYblY/edit?usp=sharing).

~~Curently waiting on packages in #3531 as `/usr/bin/perf` in `kernel-tools` and `damo` require python.~~ Rebuilding for python 3.10 in progress.

Package(s) Affected
-------------------

* pahole 1.22
* linux-kernel, kernel-tools, linux+kernel 5.15.4
* util-linux: 2.37.2
* libbpf 0.5.0
* bcc 0.22.0

+ damo[noarch] 1.0.4

Build Order
-----------

pahole linux-kernel kernel-tools linux+kernel util-linux libbpf bcc damo

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`

**Secondary Architectures**

Configured but untested.

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`

Update(s) Uploaded to Stable
----------------------------

**Primary Architectures**

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`

**Secondary Architectures**

- [ ] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`

### CODE REVIEW

File: extra-kernel/libbpf/autobuild/build [hunk 1]
```diff
@@ -6,4 +6,4 @@ make install DESTDIR="${PKGDIR}" LIBSUBDIR=lib install install_headers
 popd
 
 abinfo "Installing readme documents ..."
-install -Dm644 "${SRCDIR}"/README.md "${SRCDIR}"/src/README.rst -t "${PKGDIR}/usr/share/doc/libbpf"
+install -Dm644 "${SRCDIR}"/README.md -t "${PKGDIR}/usr/share/doc/libbpf"
```
Comment (KexyBiscuit) [inline extra-kernel/libbpf/autobuild/build]:
```suggestion
install -Dvm644 "${SRCDIR}"/README.md -t "${PKGDIR}/usr/share/doc/libbpf"
```

File: extra-kernel/linux+kernel/spec [hunk 2]
```diff
@@ -1,3 +1,4 @@
-VER=5.14.15
-DUMMYSRC=1
+VER=5.15.4
+SRCS="tbl::https://cdn.kernel.org/pub/linux/kernel/v5.x/linux-${VER/.0/}.tar.xz"
```
Comment (KexyBiscuit) [inline extra-kernel/linux+kernel/spec]: But why?

**Review by KexyBiscuit — CHANGES_REQUESTED** [review]
LGTM overall, a few changes.

File: extra-kernel/linux+kernel/spec (hunk 2 above)
Comment (cthbleachbit) [reply extra-kernel/linux+kernel/spec]: My bad. I replaced the spec file from other concrete packages.

**Review by KexyBiscuit — COMMENTED** [review]
LGTM, waiting for test results.

**Comment by cthbleachbit** [comment]
Dogfooding uptime 3 days. I suppose this is good enough for stable.

**Review by KexyBiscuit — APPROVED** [review]

**Pull request merged**